### PR TITLE
[Debug Endpoint] Add support to retrieve bolt config settings

### PR DESCRIPTION
**Pull request by Jiyuan-Bolt**
# Description
This PR adds support to retrieve bolt config settings for the debug endpoint. Sensitive information like signing secret and API keys are obscured in a format like abc***efg (only displays the first and last 3 characters)

Fixes: [ENG-341]

#changelog [Debug Endpoint] Add support to retrieve bolt config settings

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

[ENG-341]: https://boltpay.atlassian.net/browse/ENG-341

### CODE REVIEW

File: Helper/Config.php [hunk 1]
```diff
@@ -435,6 +440,22 @@ public function getAnyPublishableKey($storeId = null)
         return $this->getPublishableKeyCheckout($storeId) ?: $this->getPublishableKeyPayment($storeId);
     }
 
+	/**
```
Comment (matt-thomason) [inline Helper/Config.php]: Please add a unit test here

File: Helper/SecretObscurer.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+class SecretObscurer
+{
+	/**
+	 * Obscures the input string.
+	 *
+	 * @param string $input
+	 * @return string
+	 */
+	public static function obscure($input)
```
Comment (matt-thomason) [inline Helper/SecretObscurer.php]: Please create a test for this new file

File: Model/Api/Debug.php [hunk 3]
```diff
@@ -79,14 +111,187 @@ public function debug()
...
+		// API Key (obscured)
+		$boltSettings[] = $this->boltConfigSettingFactory->create()
+		                                                 ->setName('api_key')
+		                                                 ->setValue(SecretObscurer::obscure($this->configHelper->getApiKey()));
+		// Signing Secret (obscured)
+		$boltSettings[] = $this->boltConfigSettingFactory->create()
+		                                                 ->setName('signing_secret')
+		                                                 ->setValue(SecretObscurer::obscure($this->configHelper->getSigningSecret()));
+		// Publishable Key for Checkout (obscured)
+		$boltSettings[] = $this->boltConfigSettingFactory->create()
```
Comment (matt-thomason) [inline Model/Api/Debug.php]: Publishable keys are public - no need to obscure them.

Comment (Jiyuan-Bolt) [reply Model/Api/Debug.php]: fixed

File: Helper/SecretObscurer.php (hunk 2 above)
Comment (Jiyuan-Bolt) [reply Helper/SecretObscurer.php]: good catch, added

File: Helper/Config.php (hunk 1 above)
Comment (Jiyuan-Bolt) [reply Helper/Config.php]: added

**Comment by Jiyuan-Bolt** [comment]
> I think a centralized allConfigs() function would be most elegant, but this ships fine

Fixed.

File: Helper/Config.php [hunk 4]
```diff
@@ -316,19 +328,22 @@ class Config extends AbstractHelper
...
      *
      * @codeCoverageIgnore
      */
     public function __construct(
         Context $context,
         EncryptorInterface $encryptor,
         ResourceInterface $moduleResource,
-        ProductMetadataInterface $productMetadata
+        ProductMetadataInterface $productMetadata,
+	    BoltConfigSettingFactory $boltConfigSettingFactory
```
Comment (Jiyuan-Bolt) [inline Helper/Config.php]: Not sure why this is off here, but in my IDE it is fine

File: Helper/SecretObscurer.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	{
+		if (strlen($input) == 0) {
+			return '';
+		}
+		if (strlen($input) < 6) {
+			return '***';
+		}
+		return substr($input, 0, 3) . '***' . substr($input, -3);
```
Comment (vitaliyreznikov) [inline Helper/SecretObscurer.php]: What if also return md5($input) or something like this? We can use it to check if the key set up correctly.

Comment (vitaliyreznikov) [reply Helper/SecretObscurer.php]: We can do it someday though.

Comment (Jiyuan-Bolt) [reply Helper/SecretObscurer.php]: > We can do it someday though.

Discussed in Slack and will keep it simple for now.

File: Model/Api/Debug.php [hunk 6]
```diff
@@ -100,9 +100,8 @@ public function debug()
 		# populate platform version
 		$result->setPlatformVersion($this->productMetadata->getVersion());
 
-		$boltSettings = [];
-		$boltSettings[] = $this->boltConfigSettingFactory->create();
-		$result->setBoltConfigSettings($boltSettings);
+		# populate bolt config settings
+		$result->setBoltConfigSettings($this->configHelper->getAllConfigSettings());
```
Comment (matt-thomason) [inline Model/Api/Debug.php]: 🙏 Thanks!

**Pull request merged**